wave_lang/kernel/wave/mlir_converter/water_emitter.py

```diff
@@ -144,12 +146,13 @@ def _type_to_wave_mlir(
     raise RuntimeError(f"Unsupported wave type for MLIR conversion: {type_}")
 
 
-def _read_trace() -> tuple[CapturedTrace, WaveCompileOptions]:
-    """Reads and returns a pickled trace and options from stdin.
+def _parse_input() -> tuple[CapturedTrace, WaveCompileOptions, str]:
+    """Parses and returns the pickled trace, options, and pipeline from stdin.
 
     The input is expected to be a dill-serialized dict with keys:
     - "trace": CapturedTrace object
     - "options": WaveCompileOptions
+    - "pipeline": str
 
```

Comment:
Please add to the docstring that the pipeline is (eventually meant to be) a transform module